nixos/modules/services/logging/packetbeat.nix

```diff
@@ -0,0 +1,190 @@
+          - type: amqp
+            ports: [5672]
+          - type: cassandra
+            ports: [9042]
+          - type: dhcpv4
+            ports: [67, 68]
+          - type: dns
+            ports: [53]
+          - type: http
+            ports: [80, 8080, 8000, 5000, 8002]
+          - type: memcache
+            ports: [11211]
+          - type: mysql
+            ports: [3306,3307]
+          - type: pgsql
+            ports: [5432]
+          - type: redis
+            ports: [6379]
+          - type: thrift
+            ports: [9090]
+          - type: mongodb
+            ports: [27017]
+          - type: nfs
+            ports: [2049]
+          - type: tls
+            ports:
+              - 443   # HTTPS
+              - 993   # IMAPS
+              - 995   # POP3S
+              - 5223  # XMPP over SSL
+              - 8443
+              - 8883  # Secure MQTT
+              - 9243  # Elasticsearch
+
+        '';
+        description = ''
+          Configuration of what protocols packetbeat should gather info about.
+          See <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/configuration-protocols.html'/>
+          for the configuration options available.
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = ''
+          packetbeat.interfaces.device: any
+
+          setup.template.settings:
+            index.number_of_shards: 1
+
+          setup.kibana:
+            host: "localhost:5601"
+
+          output.elasticsearch:
+            hosts: ["localhost:9200"]
+
+          processors:
+            - # Add forwarded to tags when processing data from a network tap or mirror.
+              if.contains.tags: forwarded
+              then:
+                - drop_fields:
+                    fields: [host]
+              else:
+                - add_host_metadata: ~
+            - add_cloud_metadata: ~
+            - add_docker_metadata: ~
+        '';
+        description = "Any other configuration options you want to add";
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    warnings = optional (cfg.configFile != null && (cfg.configProtocols != null || cfg.configFlows != null || cfg.extraConfig != null)) ''
+      Setting services.packetbeat.configFile will overrule the other configuration settings in the module.
+    '';
+    assertions = [
+      {
+        assertion = (cfg.configProtocols != "" || cfg.configFlows != "") || cfg.configFile != null;
+        message = ''
+          The options services.packetbeat.configProtocols, services.packetbeat.configFlows and/or services.packetbeat.configFile should be set or else packetbeat won't do anything useful and error out.
+        '';
+      }
+    ];
+
+    systemd.services.packetbeat = {
+      description = "Packetbeat log shipper";
+      wantedBy = [ "multi-user.target" ];
+      preStart = ''
+        mkdir -p ${cfg.stateDir}/data
+        mkdir -p ${cfg.stateDir}/logs
```

Comment:
Is it possible to log to `stdout` instead so `journald` can pick this up?

Reply:
There is an option to log to stderr (-e) that could be used in ExecStart

Reply:
There is an option to log to stderr that should make it possible for journald to pick it up (-e).